libvast/vast/system/node_control.hpp

```diff
@@ -36,43 +37,11 @@ spawn_at_node(caf::scoped_actor& self, caf::actor node, Arguments&&... xs) {
   return result;
 }
 
-/// Look up components by category. Returns the first actor of each
-/// category name passed in `names`.
-/// TODO: Replace all usages of get_node_components with
-/// get_typed_node_components.
-template <class... Names>
-caf::expected<std::array<caf::actor, sizeof...(Names)>>
-get_node_components(caf::scoped_actor& self, const caf::actor& node,
-                    Names&&... names) {
-  static_assert(
-    std::conjunction_v<std::is_constructible<std::string, Names>...>,
-    "name parameter cannot be used to construct a string");
-  auto result = caf::expected{std::array<caf::actor, sizeof...(names)>{}};
-  auto labels = std::vector<std::string>{std::forward<Names>(names)...};
-  self
-    ->request(node, caf::infinite, atom::get_v, atom::label_v,
-              std::move(labels))
-    .receive(
-      [&](std::vector<caf::actor>& components) {
-        VAST_ASSERT(components.size() == sizeof...(names));
-        std::move(components.begin(), components.end(), result->begin());
-      },
-      [&](caf::error& e) { result = std::move(e); });
-  return result;
-}
-
-struct actor_cast_wrapper {
-  template <class Out, class In>
-  Out operator()(In&& in) {
-    return caf::actor_cast<Out>(std::forward<In>(in));
-  }
-};
-
 /// Look up components by their typed actor interfaces. Returns the first actor
 /// of each type passed as template parameter.
 template <class... Actors>
 caf::expected<std::tuple<Actors...>>
-get_typed_node_components(caf::scoped_actor& self, const caf::actor& node) {
+get_node_components(caf::scoped_actor& self, const caf::actor& node) {
```

Comment:
I wish we could have a better name for this function, but I can't come up with one...

Reply:
Let's not change that within this PR.